server/src/main/java/io/crate/expression/symbol/Symbols.java

```diff
@@ -170,6 +172,14 @@ public static String format(String messageTmpl, Symbol... symbols) {
         return String.format(Locale.ENGLISH, messageTmpl, formattedSymbols);
     }
 
+    public static Symbol unwrapReferenceFromCast(Symbol symbol) {
```

Comment:
Looks like in the call-sites of this we're now creating comparisons where the types don't match. Shouldn't they still be adjusted somehow? 

I'm afraid that this might bite us later if we don't do that.

Reply:
It is only used to extract primary key columns or columns which generated column expressions are including/targeting to expand queries with the generated column. So the symbol tree will not be changed, the expanded generated column will still take the original comparison into account. Do I maybe miss something?

Reply:
> Do I maybe miss something?

No, thanks for the clarification - I misunderstood what's happening.